src/Sulu/Bundle/WebsiteBundle/Command/DumpSitemapCommand.php

```diff
@@ -0,0 +1,228 @@
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Dump sitemaps.
+ */
+class DumpSitemapCommand extends ContainerAwareCommand
+{
+    /**
+     * @var SitemapProviderPoolInterface
+     */
+    private $pool;
+
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * @var string
+     */
+    private $baseDirectory;
+
+    /**
+     * @var string
+     */
+    private $defaultHost;
+
+    /**
+     * @var string
+     */
+    private $scheme = 'http';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('sulu:website:dump-sitemap')
+            ->addOption('https', null, InputOption::VALUE_NONE, 'Use https scheme for url generation.')
+            ->addOption('clear', null, InputOption::VALUE_NONE, 'Delete all file before start.');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->pool = $this->getContainer()->get('sulu_website.sitemap.pool');
+        $this->filesystem = new Filesystem();
+
+        $this->environment = $this->getContainer()->getParameter('kernel.environment');
+        $this->baseDirectory = $this->getContainer()->getParameter('sulu_website.sitemap.dump_dir');
+        $this->defaultHost = $this->getContainer()->getParameter('sulu_website.sitemap.default_host');
+
+        if ($input->getOption('https')) {
+            $this->scheme = 'https';
+        }
+
+        if ($input->getOption('clear')) {
+            $this->filesystem->remove(rtrim($this->baseDirectory, '/') . '/' . $this->scheme);
+        }
+
+        $output->writeln('Start dumping "sitemap.xml" files:');
+
+        $webspaceManager = $this->getContainer()->get('sulu_core.webspace.webspace_manager');
+        foreach ($webspaceManager->getPortalInformations($this->environment) as $portalInformation) {
+            $this->dumpSitemap($portalInformation);
+        }
+    }
+
+    /**
+     * Dump sitemaps for portal-information.
+     *
+     * @param PortalInformation $portalInformation
+     */
+    private function dumpSitemap(PortalInformation $portalInformation)
+    {
+        if (-1 !== strpos($portalInformation->getUrl(), '{host}')) {
+            if (!$this->defaultHost) {
+                return;
+            }
+
+            $portalInformation->setUrl(str_replace('{host}', $this->defaultHost, $portalInformation->getUrl()));
+        }
+
+        if (!$this->pool->hasIndex()) {
+            return $this->dumpFile(
+                '/' . $portalInformation->getUrl() . '/sitemap.xml',
+                $this->renderProviderSitemap($this->pool->getFirstAlias(), $portalInformation)
```

Comment:
Or you could use the already existing `getProviders` function, couldn't you? You are also already loading a provider in the other function, so that would probably make sense.